src/addDefaultProps.coffee

```diff
@@ -0,0 +1,18 @@
+import {isFunction} from './util/helpers'
+import addProps from './addProps'
+
+addDefaultProps = (createDefaults) ->
+  addProps (props) ->
+    defaults = if isFunction createDefaults
+      createDefaults props
+    else
+      createDefaults
+
+    newProps = {}
+
+    for key, val of defaults when not props[key]
```

Comment:
`when not props[key]` is a truthy test, whereas `when not props[key]?` would be a nullish test

Which points at a missing test case for a falsy but not nullish prop value (which I think you had previously)